keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1458 @@
+
+To support pod-level resource specifications for hugepages, Kubernetes will need to adjust how it handles hugetlb cgroups. Unlike memory, where an unset limit 
+means unlimited, an unset hugetlb limit is the same as setting it to 0.
+
+With the proposed changes, hugepages-2Mi and hugepages-1Gi will be added to the pod-level resources section, alongside CPU and memory. The hugetlb cgroup for the
+pod will then directly reflect the pod-level hugepage limits, rather than using an aggregated value from container limits. When scheduling, the scheduler will 
+consider hugepage requests at the pod level to find nodes with enough available resources.
+
+
+#### [Scoped for Beta] Topology Manager
+
+Note: This section includes only high level overview; Design details will be added in Beta stage.
+
+
+* (Tentative) Only pod level scope for topology alignment will be supported if pod level requests and limits are specified without container-level requests and limits.
+* The pod level scope for topology aligntment will consider pod level requests and limits instead of container level aggregates.
+* The hint providers will consider pod level requests and limits instead of container level aggregates.
+
+
+#### [Scoped for Beta] Memory Manager
+
+Note: This section includes only high level overview; Design details will be added in Beta stage.
+
+With the introduction of pod-level resource specifications, the Kubernetes Memory Manager will evolve to track and enforce resource limits at both the pod and container levels. It will need to aggregate memory usage across all containers within a pod to calculate the pod's total memory consumption. The Memory Manager will then enforce the pod-level limit as the hard cap for the entire pod's memory usage, preventing it from exceeding the allocated amount.  While still maintaining container-level limit enforcement, the Memory Manager will need to coordinate with the Kubelet and eviction manager to make decisions about pod eviction or individual container termination when the pod-level limit is breached.
+
+
+#### [Scoped for Beta] CPU Manager
+
+Note: This section includes only high level overview; Design details will be added in Beta stage.
+
+With the introduction of pod-level resource specifications, the CPU manager in Kubernetes will adapt to manage CPU requests and limits at the pod level rather than solely at the container level. This change means that the CPU manager will allocate and enforce CPU resources based on the total requirements of the entire pod, allowing for more flexible and efficient CPU utilization across all containers within a pod. The CPU manager will need to ensure that the aggregate CPU usage of all containers in a pod does not exceed the pod-level limits.
+
+#### [Scoped for Beta] In Place Pod Resize
+
+TBD. Do not review for the alpha stage.
+
+#### [Scoped for Beta] VPA
+
+TBD. Do not review for the alpha stage.
+
+#### [Scoped for Beta] Cluster Autoscaler
```

Comment:
so what will be behavior in alpha? Will CA assume that pod doesn't need resorurces and will not create new nodes for it?

Reply:
I think that's actually a desired behavior, but we need to check if that's true @ndixita

Reply:
let's mention it than. If enabling of this feature while in alpha will cause CA failures, it needs to be at least mentioned.

Reply:
@ndixita please add a sentence to describe the CA behavior when the alpha feature is enabled.